evol/logger.py

```diff
@@ -0,0 +1,86 @@
+import datetime as dt
+import os
+import json
+
+
+class BaseLogger():
+    def __init__(self, file=None):
+        self.file = file
+        if self.file is not None:
+            if not os.path.exists(os.path.split(file)[0]):
+                raise RuntimeError(f"path to file {os.path.split(file)[0]} does not exist!")
+
+    def log(self, population, **kwargs):
+        """
+        The logger method of the Logger object determines what will be logged. 
+        :param population: 
+        :return: generator of strings to be handled
+        """
+        values = ','.join(kwargs.values())
+        for i in population:
+            string = f'{dt.datetime.now()},{population.id},{i.id},{i.fitness}' + values
+            self.handle(string)
```

Comment:
I think it is much better that we handle the `handle` internally, such that the `Population` only makes a single call. I think we may take it even one step further: have `log()` call two methods:

```
def log(self, population, **kwargs):
    lines = self.format(population, **kwargs)
    self.handle(lines)
```

This way a user can easily inherit from our loggers and overwrite `format`. He/she doesn't need to think about calling `handle`; only to return what needs to be logged.